editor/editor_data.cpp

```diff
@@ -48,8 +48,16 @@ void EditorHistory::_cleanup_history() {
 			if (!history[i].path[j].ref.is_null())
 				continue;
 
-			if (ObjectDB::get_instance(history[i].path[j].object))
-				continue; //has isntance, try next
+
+			Object* obj = ObjectDB::get_instance(history[i].path[j].object);
+			if(obj)
```

Comment:
Style issues, it should be `if (obj) {`. See https://travis-ci.org/godotengine/godot/jobs/326249268 and http://docs.godotengine.org/en/latest/community/contributing/code_style_guidelines.html

While you amend the commit, might be worth also editing the commit log to describe what actual engine bug is being fixed (editor history not being cleaned up properly) instead of the symptom (deleted nodes are "hanging"). So something like:
```
Clean the EditorHistory on SceneTree node deletion

Fixes #15416.
```

Reply:
This said, congrats on your first PR, it looks fine overall code-wise :)
You may be interested in reading (or skimming through, depending on your experience) http://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html too.